pkg/awsvpc/parser.go

```diff
@@ -0,0 +1,344 @@
+	}
+
+	err = rc.getSGconfig(res, shouldSkipVpcIds, netIntfToSGs)
+	if err != nil {
+		return nil, err
+	}
+
+	printVPCConfigs(res)
+
+	return res, nil
+}
+
+func (rc *AWSresourcesContainer) getVPCconfig(
+	res *vpcmodel.MultipleVPCConfigs,
+	skipByVPC map[string]bool) error {
+	for _, vpc := range rc.VpcsList {
+		if skipByVPC[*vpc.VpcId] {
+			continue // skip vpc not specified to analyze
+		}
+
+		vpcNodeSet, err := commonvpc.NewVPC(*vpc.VpcId, *vpc.VpcId, "", nil, nil)
+		if err != nil {
+			return err
+		}
+
+		newVPCConfig := commonvpc.NewEmptyVPCConfig()
+		newVPCConfig.UIDToResource[vpcNodeSet.ResourceUID] = vpcNodeSet
+		newVPCConfig.VPC = vpcNodeSet
+		res.SetConfig(vpcNodeSet.ResourceUID, newVPCConfig)
+	}
+	if len(res.Configs()) == 0 {
+		return errors.New("could not find any VPC to analyze")
+	}
+	return nil
+}
+
+func (rc *AWSresourcesContainer) getInstancesConfig(
+	subnetIDToNetIntf map[string][]*commonvpc.NetworkInterface,
+	netIntfToSGs map[string][]types.GroupIdentifier,
+	res *vpcmodel.MultipleVPCConfigs,
+	skipByVPC map[string]bool,
+) error {
+	for _, instance := range rc.InstancesList {
+		vpcUID := *instance.VpcId
+		if skipByVPC[vpcUID] {
+			continue
+		}
+		vpc, err := commonvpc.GetVPCObjectByUID(res, vpcUID)
+		if err != nil {
+			return err
+		}
+		vsiNode, err := commonvpc.NewVsi(*instance.InstanceId, *instance.InstanceId, *instance.Placement.AvailabilityZone, vpc, res)
+		if err != nil {
+			return err
+		}
+		vpcConfig := res.Config(vpcUID)
+		vpcConfig.NodeSets = append(vpcConfig.NodeSets, vsiNode)
+		vpcConfig.UIDToResource[vsiNode.ResourceUID] = vsiNode
+		for j := range instance.NetworkInterfaces {
+			netintf := instance.NetworkInterfaces[j]
+			// netintf has no CRN, thus using its ID for ResourceUID
+			intfNode, err := commonvpc.NewNetworkInterface(*netintf.NetworkInterfaceId, *netintf.NetworkInterfaceId,
+				*instance.Placement.AvailabilityZone, *netintf.PrivateIpAddress, *instance.InstanceId, vpc)
+			if err != nil {
+				return err
+			}
+			netIntfToSGs[*netintf.NetworkInterfaceId] = netintf.Groups
+			vpcConfig.Nodes = append(vpcConfig.Nodes, intfNode)
+			vpcConfig.UIDToResource[intfNode.ResourceUID] = intfNode
+			vsiNode.VPCnodes = append(vsiNode.VPCnodes, intfNode)
+			subnetID := *netintf.SubnetId
+			if _, ok := subnetIDToNetIntf[subnetID]; !ok {
+				subnetIDToNetIntf[subnetID] = []*commonvpc.NetworkInterface{}
+			}
+			subnetIDToNetIntf[subnetID] = append(subnetIDToNetIntf[subnetID], intfNode)
+		}
+	}
+	return nil
+}
+
+func (rc *AWSresourcesContainer) getSubnetsConfig(
+	res *vpcmodel.MultipleVPCConfigs,
+	subnetNameToNetIntf map[string][]*commonvpc.NetworkInterface,
+	skipByVPC map[string]bool,
+) (vpcInternalAddressRange map[string]*ipblock.IPBlock, err error) {
+	vpcInternalAddressRange = map[string]*ipblock.IPBlock{}
+	for vpcUID := range res.Configs() {
+		vpcInternalAddressRange[vpcUID] = nil
+	}
+	for _, subnet := range rc.SubnetsList {
+		if skipByVPC[*subnet.VpcId] {
```

Comment:
Is this needed here? didn't we already skipped in the `VPC` level?

Reply:
we should skip subnets that are associated with this vpcID